<h2>Soal 1: Membuat Objek DataFrames</h2>

- Buatlah objek dataframes dari objek dictionary, Bebas!

In [15]:
# Membuat dataframes dari dictionary
import pandas as pd

df = pd.DataFrame({'Nama' : ['dean','roni'], 'Umur' : [21,19]})
print(df.index('Nama'))

TypeError: 'RangeIndex' object is not callable

<h2>Soal 2: Membaca data dari file</h2>

Download file bernama [diamonds.csv](https://drive.google.com/uc?export=download&id=1qQiM9utQUThTPq8jk8stPMh_y2zfqYOd), kemudian baca data tersebut dan assign ke dalam suatu variabel kemudian print.

In [9]:
# Membaca data dan memasukannya kedalam suatu variabel
df_from_csv = pd.read_csv('diamonds-1.csv')
print(df_from_csv)


       Unnamed: 0  carat        cut color clarity  depth  table  price     x  \
0               1   0.23      Ideal     E     SI2   61.5   55.0    326  3.95   
1               2   0.21    Premium     E     SI1   59.8   61.0    326  3.89   
2               3   0.23       Good     E     VS1   56.9   65.0    327  4.05   
3               4   0.29    Premium     I     VS2   62.4   58.0    334  4.20   
4               5   0.31       Good     J     SI2   63.3   58.0    335  4.34   
...           ...    ...        ...   ...     ...    ...    ...    ...   ...   
53935       53936   0.72      Ideal     D     SI1   60.8   57.0   2757  5.75   
53936       53937   0.72       Good     D     SI1   63.1   55.0   2757  5.69   
53937       53938   0.70  Very Good     D     SI1   62.8   60.0   2757  5.66   
53938       53939   0.86    Premium     H     SI2   61.0   58.0   2757  6.15   
53939       53940   0.75      Ideal     D     SI2   62.2   55.0   2757  5.83   

          y     z  
0      3.98  2.43  

Expected Result:

![alt text](https://drive.google.com/uc?id=1BvgM5mJEVfj0fllrjkBfQwn0QIFkry_t)

<h2>Soal 3: DataFrame vs Series</h2>

- Apa perbedaan Series dengan list dan dictionary
- Apa perbedaan dataframe dan series
- Buatlah suatu data series

Jawab: series merupakan gabungan antara list dan dictionary, series memiliki keys seperti halnya pada dictionary, dan dapat mengakses value menggunakan indexing seperti halnya pada list.

perbedaan series dan index adalah, series merupakan onjek satu dimensi, sedangkan dataframe merupakan obejk dua dimensi

In [12]:
# code here
ds = pd.Series([1,2,3,4,5], index=['a','b','c','d','e'])
print(ds)

a    1
b    2
c    3
d    4
e    5
dtype: int64


<h2>Soal 4: Pemeriksaan Data Sederhana</h2>

Di soal no 2 kamu telah membaca data tentang diamonds. pada soal ini cobalah suatu metode dari pandas untuk mengetahui beberapa data pertama dan beberapa data terakhir. kemudian sebutkan nilai dari:
- baris pertama dari column price
- baris terakhir dari column color

tunjukan cara kalian menemukan nilai tersebut di block di bawah ini.

In [19]:
# code here
first_data, last_data = df_from_csv.head(1), df_from_csv.tail(1)
print(first_data["price"])
print(last_data["color"])


0    326
Name: price, dtype: int64
53939    D
Name: color, dtype: object


<h2>Soal 5: Deskriptif Statistik</h2>

Jelaskan apa itu deskriptis statistik!

Jawab: deskriptis statistik merupakan metode untuk menyajikan kesimpulan statistik dari label/kolom pada data

<h2>Soal 6: Practice Deskriptif Statistik</h2>

- Gunakan suatu metode dari pandas untuk mengetahui deskriptif statistik dari suatu data
- Berapa nilai rata-rata dari column price
- Berapa nilai standar deviasi dari column depth
- Berapa nilai maximum dari column carat

In [31]:
# code here
print("METODE DESCRIBE()")
print(df_from_csv.describe(), "\n")

mean_price, std_depth, max_carat = df_from_csv["price"], df_from_csv["depth"], df_from_csv["carat"]
print("NILAI RATA-RATA DARI KOLOM PRICE")
print(mean_price.mean(), "\n")
print("NILAI STANDAR DEVIASI DARI KOLOM DEPTH")
print(std_depth.std(), "\n")
print("NILAI TERBESAR DARI KOLOM CARAT")
print(max_carat.max(), "\n")


METODE DESCRIBE()
         Unnamed: 0         carat         depth         table         price  \
count  53940.000000  53940.000000  53940.000000  53940.000000  53940.000000   
mean   26970.500000      0.797940     61.749405     57.457184   3932.799722   
std    15571.281097      0.474011      1.432621      2.234491   3989.439738   
min        1.000000      0.200000     43.000000     43.000000    326.000000   
25%    13485.750000      0.400000     61.000000     56.000000    950.000000   
50%    26970.500000      0.700000     61.800000     57.000000   2401.000000   
75%    40455.250000      1.040000     62.500000     59.000000   5324.250000   
max    53940.000000      5.010000     79.000000     95.000000  18823.000000   

                  x             y             z  
count  53940.000000  53940.000000  53940.000000  
mean       5.731157      5.734526      3.538734  
std        1.121761      1.142135      0.705699  
min        0.000000      0.000000      0.000000  
25%        4.710000 

Jawab: <br>
NILAI RATA-RATA DARI KOLOM PRICE = 3932.799721913237 <br>
NILAI STANDAR DEVIASI DARI KOLOM DEPTH = 1.4326213188337733 <br>
NILAI TERBESAR DARI KOLOM CARAT = 5.01 